server/game/cards/06-CotE/UnveiledDestiny.js

```diff
@@ -0,0 +1,16 @@
+const DrawCard = require('../../drawcard.js');
+const AbilityDsl = require('../../abilitydsl.js');
+
+class UnveiledDestiny extends DrawCard {
+    setupCardAbilities() {
+        this.persistentEffect({
+            condition: context => context.source.parent.isAttacking() && context.player.role.getElement() !== [],
+            match: ring => ring.contested,
+            effect: context => context.player.role.getElement().forEach(element => AbilityDsl.effects.addElement(element))
```

Comment:
I think `map` is what you mean here.... this should be `context => context.player.role.getElement().map(element => AbilityDsl.effects.addElement(element))`

Reply:
Looking at this again, it's more tricky that I first thought.  You can't pass a function to `effect`.  There are two ways to implement this I think:
1) Create 5 persistent effects, one for each element, which condition on whether that element is on the controllers role card. The problem with doing it this way is that it's only a single effect on the physical card
2) Change the way the `addElement` effect works.  It needs to be changed to be `flexible` instead of `static` and then changed so that it takes an array of elements as well as a single element.  Then you can pass the effect a function which returns `getElement()`

If you want to go with option 2, but you'd prefer not to delve into the guts of the engine, let me know and I'll do that bit

Reply:
I'll definitely give option 2 a shot. Thanks for the feedback

Reply:
so with my latest commit I think getElement should be able to an array of elements. so now that its flexible it will run `EffectBuilder.ring.dynamic` I changed that function so it will map the array given so it will return an array of `RingEffect` classes instead of just one. 

then the `presistentEffect` function on the `UnveilingDestiny` class will add this array of `RingEffect` classes to the engine via the `addEffectToEngine` method which is defined of the `EffectSource` class. this method already handles arrays of effects as well as single effect, so I think it should work.

I am going to start writing tests for this, let me know if I have an incorrect understand of the working of the engine.